src/interactable_spring.c

```diff
@@ -0,0 +1,309 @@
+static const u8 gUnknown_080D53D0[SPRINGTYPE_COUNT]
+    = { 14, 15, 16, 17, 18, 19, 20, 21, 18, 19 };
+
+static const u16 sSpring_MusicPlant_Soundeffects[5]
+    = { SE_MUSIC_PLANT_SPRING_1, SE_MUSIC_PLANT_SPRING_2, SE_MUSIC_PLANT_SPRING_3,
+        SE_MUSIC_PLANT_SPRING_4, MUS_DUMMY };
+
+// Finished, but doesn't match.
+static void initSprite_Interactable_Spring(u8 springType, Interactable *inEntity,
+                                           u16 spriteRegionX, u16 spriteRegionY,
+                                           u8 spriteY)
+{
+    s16 springKind = SPRING_KIND_NORMAL;
+    struct Task *t = TaskCreate(&Task_Interactable_Spring, sizeof(Sprite_Spring), 0x2010,
+                                0, &TaskDestructor_Interactable_Spring);
+    Sprite_Spring *spring = TaskGetStructPtr(t);
+    struct UNK_0808B3FC_UNK240 *main = &spring->main;
+
+    spring->base.regionX = spriteRegionX;
+    spring->base.regionY = spriteRegionY;
+    spring->base.entity = inEntity;
+    spring->base.spriteX = inEntity->x;
+    spring->base.spriteY = spriteY;
+
+    main->unk16 = SpriteGetScreenPos(inEntity->x, spriteRegionX);
+    main->unk18 = SpriteGetScreenPos(inEntity->y, spriteRegionY);
+    SET_SPRITE_INITIALIZED(inEntity);
+
+    main->unk1A = 0x480;
+    main->unk8 = springKind;
+    main->unk14 = springKind;
+    main->unk1C = springKind;
+
+    main->unk21 = 0xFF;
+    main->unk22 = 0x10;
+    main->unk25 = 0;
+    main->unk28 = -1;
+    main->unk10 = 0x2200;
+
+    if (LEVEL_TO_ZONE(gCurrentLevel) == ZONE_3)
+        springKind = SPRING_KIND_MUSIC_PLANT;
+
+    if (LEVEL_TO_ZONE(gCurrentLevel) == ZONE_6)
+        springKind = SPRING_KIND_TECHNO_BASE;
+
+    if (((s16)springKind != SPRING_KIND_MUSIC_PLANT) || ((springType / 2) != 0)) {
+        u16 tileCount = sSpringAnimationData[springKind][springType][2];
+        main->unk4 = VramMalloc(tileCount);
+    } else {
+        main->unk4 = (void *)(OBJ_VRAM0 + 0x2980);
+    }
+
+    main->unkA = sSpringAnimationData[springKind][springType][0];
+    main->unk20 = sSpringAnimationData[springKind][springType][1];
+
+    main->unk10 |= sSpringAnimationData[springKind][springType][3];
+    spring->unk3D = springType;
+    spring->unk3E = inEntity->data[0] & 0x3;
+    sub_8004558(main);
+}
+
+// @HACK
+#define TO_TYPE(type, p, index) (((type *)(p))[index])
+
+// TODO: make static
+static void Task_Interactable_Spring(void)
+{
+    Sprite_Spring *spring = TaskGetStructPtr(gCurTask);
+    struct UNK_0808B3FC_UNK240 *main = &spring->main;
+    Interactable *entity = spring->base.entity;
+
+    if (sub_800E490(main, entity, spring, &gPlayer) != 0) {
+        gCurTask->main = &sub_800E3D0;
+        main->unk20++;
+
+        if ((LEVEL_TO_ZONE(gCurrentLevel) == ZONE_3 && (spring->unk3D / 2) == 0))
+            main->unk4 = (void *)(OBJ_VRAM0 + 0x2B00);
+    }
+
+#ifndef NON_MATCHING
+    if (IS_OUT_OF_CAM_RANGE(main->unk16, TO_TYPE(s16, main, 12)))
```

Comment:
`unkC` of `UNK_0808B3FC_UNK240` is a pointer to `UNK_0808B3FC_UNK240_UNKC` which is only really used in `sprite_3.c`

You may find the use of `unk1` can be written as `unkC->unk1 >> 8` and then you can convert this to `main->unk0` (With `unk0` being `s16`)

Reply:
Oh wait, this is `unk18` 🤔

Reply:
I tried that before it matched, will have a look

Reply:
The code you have should produce the same result :/

Reply:
Yeah, it worked after matching! :)